lib/mayaUsd/resources/scripts/mayaUsdAddMayaReference.py

```diff
@@ -49,47 +53,158 @@ def createPrimAndAttributes(stage, primPath, mayaReferencePath, mayaNamespace, m
 
     return prim
 
+def getDefaultGroupPrimName(parentPrim, mayaNamespace):
+    # "The name of the reference file" + "RN" + "group" for example 'sphereRNgroup'.
+    # If a prim with this name already exists, then add a numeric suffix after "RN".
+    startName = mayaNamespace + "RNgroup"
+    checkName = mayaUsd.ufe.uniqueChildName(parentPrim, startName)
+    index = 1
+    while (checkName != startName):
+        startName = "%sRN%dgroup" % (mayaNamespace, index)
+        checkName = mayaUsd.ufe.uniqueChildName(parentPrim, startName)
+        index += 1
+    return startName
+
 # Adapted from testMayaUsdSchemasMayaReference.py.
 def createMayaReferencePrim(ufePathStr, mayaReferencePath, mayaNamespace, 
                             mayaReferencePrimName = kDefaultMayaReferencePrimName,
-                            variantSetName = kDefaultVariantSetName,
-                            variantName = kDefaultVariantName,
+                            groupPrim = None, variantSet = None,
```

Comment:
Made these two input vars optional. Variant should never have had a default value. That caused a variant to always be created (which is not what we want - it should only be when "Define in Variant" is checked.

Added  a bunch of doc below for each param and the new return value (for error checking from test).